### PR TITLE
fix: add SR to Pagination items (#13537)

### LINKED ISSUE #13537: Search Results Pagination requires meaningful narration
**Describe the bug**
- Pagination requires meaningful labels.
- Previous/Next Arrows and page numbers need more narration context.

**Tell us the version of Spartacus**
- Library version: 4.0.0.

**To Reproduce**
1. Go to Digital Cameras menu, select option "Shop all Digital Cameras" to get several result pages.
2. Open any Screen Reader (VO, JAWS, NVDA)
3. Tab to Pagination
4. Hear missing context.

**Expected behavior**
- Narration has more detail. eg. "Product search pagination, Page 1 selected, link _[tab to end]_ Next Page, link" 

**Screenshots**
Screen Shot 2021-08-05 at 7 04 49 PM

### PR DESCRIPTION
**Pull request by BenjaminEmiliani**
Closes #13537

### CODE REVIEW

File: projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts [hunk 1]
```diff
@@ -61,6 +61,21 @@ export class PaginationComponent {
...
+   * @param label
+   * @param item
+   * @returns
+   */
+  getAriaLabel(label: string, type: PaginationItemType) {
+    if (type === PaginationItemType.PAGE) {
+      return `${type} ${label}`;
+    }
+
+    return `${type} ${PaginationItemType.PAGE}`;
```
Comment (Zeyber) [inline projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts]: Not necessary but if you wanted to shorten this code

```suggestion
    return type === PaginationItemType.PAGE
      ? `${type} ${label}`
      : `${type} ${PaginationItemType.PAGE}`;
```

File: projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts [hunk 2]
```diff
@@ -61,6 +61,21 @@ export class PaginationComponent {
     );
   }
 
+  /**
+   * Format aria-label based on pagination page type.
+   *
+   * @param label
+   * @param item
```
Comment (Zeyber) [inline projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts]: This label `item` doesn't match the input param.

File: projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts [hunk 3]
```diff
@@ -61,6 +61,21 @@ export class PaginationComponent {
...
+   * @returns
+   */
+  getAriaLabel(label: string, type: PaginationItemType) {
```
Comment (Zeyber) [inline projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts]: Good practice to always have the return type.

```suggestion
  getAriaLabel(label: string, type: PaginationItemType): string {
```

File: projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts [hunk 4]
```diff
@@ -61,6 +61,21 @@ export class PaginationComponent {
     );
   }
 
+  /**
+   * Format aria-label based on pagination page type.
+   *
+   * @param label
+   * @param item
+   * @returns
```
Comment (Zeyber) [inline projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts]: Returns what?

**Review by Zeyber — CHANGES_REQUESTED** [review]

File: projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts (hunk 1 above)
Comment (BenjaminEmiliani) [reply projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts]: Changed

File: projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts (hunk 3 above)
Comment (BenjaminEmiliani) [reply projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts]: Fixed

File: projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts (hunk 2 above)
Comment (BenjaminEmiliani) [reply projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts]: Fixed

File: projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts (hunk 4 above)
Comment (BenjaminEmiliani) [reply projects/storefrontlib/shared/components/list-navigation/pagination/pagination.component.ts]: Fixed

**Review by developpeurweb — APPROVED** [review]
Works well with JAWS and NVDA.

**Review by Zeyber — APPROVED** [review]

**Pull request merged**